test/grouping.jl

```diff
@@ -2,44 +2,124 @@ module TestGrouping
+    f = [sum]
+    cw = colwise(f, dt)
+    answer = NullableArray([20 12 -0.4283098098931877])
+    @test isa(cw, NullableArray{Any, 2})
+    @test size(cw) == (length(f),ncol(dt))
+    @test isequal(cw, answer)
+
+    f = [sum, minimum]
+    cw = colwise(f, nullfree)
+    answer = reshape([55, 1], (2,1))
+    @test isa(cw, Array{Int, 2})
+    @test size(cw) == (length(f), ncol(nullfree))
+    @test cw == answer
+
+    f = [NullableArray]
+    cw = colwise(f, nullfree)
+    answer = reshape([NullableArray(1:10)], (1,1))
+    @test isa(cw, Array{NullableArray{Int,1},2})
+    @test size(cw) == (length(f), ncol(nullfree))
+    @test isequal(cw, answer)
+
+    @test_throws MethodError colwise(["Bob", :Susie], DataTable(A = 1:10, B = 11:20))
+
+    # colwise(::Tuple{<:Function}, ::AbstractDataTable)
+    f = (sum, length)
+    cw = colwise(f, dt)
+    answer = Any[Nullable(20) Nullable(12) Nullable(-0.4283098098931877);
+                 8            8            8                            ]
+    @test isa(cw, Array{Any, 2})
+    @test size(cw) == (length(f), ncol(dt))
+    @test isequal(cw, answer)
+
+    cw = colwise(f, nullfree)
+    answer = reshape([55, 10], (2,1))
+    @test isa(cw, Array{Int, 2})
+    @test size(cw) == (length(f), ncol(nullfree))
+    @test cw == answer
+
+    f = (CategoricalArray, NullableArray)
+    cw = colwise(f, nullfree)
+    answer = reshape([CategoricalArray(1:10), NullableArray(1:10)], (length(f), ncol(nullfree)))
+    @test isa(cw, Array{AbstractVector,2})
+    @test size(cw) == (length(f), ncol(nullfree))
+    @test isequal(cw, answer)
+
+    @test_throws MethodError colwise(("Bob", :Susie), DataTable(A = 1:10, B = 11:20))
+
+    # colwise(::Function, ::AbstractDataTable)
+    f = sum
+    cw = colwise(f, dt)
+    answer = NullableArray([20, 12, -0.4283098098931877])
+    @test isa(cw, NullableArray{Any, 1})
+    @test size(cw) == (ncol(dt),)
+    @test isequal(cw, answer)
+
+    cw = colwise(f, nullfree)
+    answer = [55]
+    @test isa(cw, Array{Int, 1})
+    @test size(cw) == (ncol(nullfree),)
+    @test cw == answer
+
+    # colwise on GroupedDataTables
+    gd = groupby(DataTable(A = [:A, :A, :B, :B], B = 1:4), :A)
+    @test colwise(length, gd) == [[2,2], [2,2]]
+    @test colwise([length], gd) == [[2 2], [2 2]]
+    @test colwise((length), gd) == [[2,2],[2,2]]
+
+    # map magic
+    cw = map(colwise(sum), (nullfree, dt))
+    answer = ([55], NullableArray(Any[20, 12, -0.4283098098931877]))
+    @test isequal(cw, answer)
+
+    cw = map(colwise((sum, length)), (nullfree, dt))
+    answer = (reshape([55, 10], (2,1)),
+              Any[Nullable(20) Nullable(12) Nullable(-0.4283098098931877);
+                  8            8            8                            ])
+    @test isequal(cw, answer)
+
+    cw = map(colwise([sum, length]), (nullfree, dt))
+    @test isequal(cw, answer)
 
-    f(dt) = DataTable(cmax = maximum(dt[:c]))
+    cols = [:a, :b]
+    fxn(dt) = DataTable(cmax = maximum(dt[:c]))
```

Comment:
I'd rather use a different name in the new code. Better avoid modifying unrelated code.